pkg/util/cloudfoundry/bbscache.go

```diff
@@ -192,39 +212,83 @@ func (bc *BBSCache) readData() {
 	bc.Lock()
 	defer bc.Unlock()
 	log.Debug("Data from BBS API read successfully, refreshing the cache")
-	bc.actualLRPs = actualLRPs
+	bc.actualLRPsByAppGUID = actualLRPsByAppGUID
+	bc.actualLRPsByCellID = actualLRPsByCellID
 	bc.desiredLRPs = desiredLRPs
 	bc.lastUpdated = time.Now()
 	bc.pollSuccesses++
 }
 
-func (bc *BBSCache) readActualLRPs() (map[string][]ActualLRP, error) {
-	actualLRPs := map[string][]ActualLRP{}
+func (bc *BBSCache) readActualLRPs() (map[string][]*ActualLRP, map[string][]*ActualLRP, error) {
+	actualLRPsByAppGUID := map[string][]*ActualLRP{}
+	actualLRPsByCellID := map[string][]*ActualLRP{}
 	actualLRPsBBS, err := bc.bbsAPIClient.ActualLRPs(bc.bbsAPIClientLogger, models.ActualLRPFilter{})
 	if err != nil {
-		return actualLRPs, err
+		return actualLRPsByAppGUID, actualLRPsByCellID, err
 	}
 	for _, lrp := range actualLRPsBBS {
 		alrp := ActualLRPFromBBSModel(lrp)
-		if lrpList, ok := actualLRPs[alrp.AppGUID]; ok {
-			actualLRPs[alrp.AppGUID] = append(lrpList, alrp)
+		if _, ok := actualLRPsByAppGUID[alrp.AppGUID]; ok {
+			actualLRPsByAppGUID[alrp.AppGUID] = append(actualLRPsByAppGUID[alrp.AppGUID], &alrp)
+			if _, ok := actualLRPsByCellID[alrp.CellID]; ok {
```

Comment:
It looks like this block is the same here and then in the `else` clause below. Maybe you should extract it and put it just once after this condition?

Reply:
doh, indeed, thanks !